src/product/commands/products/create/impl/create-product.command.ts

```diff
@@ -3,6 +3,6 @@ import { CreateProductDto } from 'src/product/dto/product/create/create-product.
 export class CreateProductCommand {
   constructor(
     public readonly product: CreateProductDto,
-    public readonly files: Array<Express.Multer.File>,
+    public readonly files: Array<Express.Multer.File> | undefined,
```

Comment:
Multer is an infrastructure tool that helps you processing files from the request. Command it's more a domain layer. And in your case your domain layer knows about the infrastructure layer. This is not critical for your size of the project, but in the big systems it's not allowed.

To prevent such cases you need create a type/interface of the arguments of your CreateProductCommand constructor.

Reply:
i don't understand it. Array<Express.Multer.File> is just type from box (nest js). Why can't I use this? Because we may change the way of uploading  files in the future?